code/script.js

```diff
@@ -218,108 +256,233 @@ const generateBoard = () => {
-  // const value =
+  const category = questions.options[questions.selectedIndex].parentNode.label;
+  const value = questions.options[questions.selectedIndex].value;
 
   currentQuestion = {
     category: category,
-    // value: value
-  }
-}
+    value: value,
+  };
+};
 
-// This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentQuestion;
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
+  // Refactored version
+  let keep = Array.isArray(secret[category]) ? secret[category].includes(value) : secret[category] === value;
+  // To understnad this refactoring first check the links below:
+  // I have a ternary operator syntax which is a bit confusing but suits for current purpose so u can read about it here:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator
+  // The Array.isArray() method determines whether the passed value is an Array. It returns a Boolean.
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray
+  // In current case if secret[category] is an Array it will continue to this evaluation operation:
+  // secret[category].includes(value)
+  // and return a result of the evaluation in boolean format. In other case if the secret[category]
+  // is not an Array it will return false and continue to
+  // evaluate secret[category] === value , and return a boolean for this evaluation.
 
-  } else if (category === 'accessories' || category === 'other') {
+  //  This code is my initial attempt and is a longer version of 286 line and also works perfectly fine.
+  //   let keep;
+  //   if (category === 'hair' || category === 'eyes') {
+  //     if (secret.hair === value || secret.eyes  === value) {
+  //       keep = true;
+  //     } else {
+  //       keep = false;
+  //     }
+  //   } else if (category === 'accessories' || category === 'other') {
+  //     if (secret.accessories.includes(value) || secret.other.includes(value)=== value) {
+  //       keep = true;
+  //     } else {
+  //       keep = false;
+  //     }
+  //   } else {
+  //     if (secret.gender === value) {
+  //       keep = true;
+  //     } else {
+  //       keep = false;
+  //     }
+  //   }
+  filterCharacters(keep);
+};
 
-  }
-}
+// Creation of customAlert (with use of boorstrap for styling)
+// Thats helped a lot to solve an issue i had https://stackoverflow.com/questions/27079598/error-failed-to-execute-appendchild-on-node-parameter-1-is-not-of-type-no?rq=1
+// this one was also educational: https://developer.mozilla.org/en-US/docs/Web/API/Element/prepend
+const customAlert = (text) => {
```

Comment:
Thanks again for linking to the stackoverflows you used in your comments so that we can all check them out!

Reply:
My pleasure ) , it is also a cool way to refresh for myself  what I was doing in the particular part of the code and why